samples/javascript/E3_GetIsClear/index.js

```diff
@@ -1,37 +1,32 @@
-const request = require("request");
+const request = require("request-promise-native");
 
 const clearWeatherConditions = ['Overcast', 'Clear', 'Partly Cloudy', 'Mostly Cloudy', 'Scattered Clouds'];
 
-module.exports = function (context, location) {
-    getCurrentConditions(location)
-        .then(function (data) {
-            const isClear = clearWeatherConditions.includes(data.weather);
-            context.done(null, isClear);
-        })
-        .catch(function (err) {
-            context.log(`E3_GetIsClear encountered an error: ${err}`);
-            context.done(err);
-        });
-};
+module.exports = async function (context, location) {
+    try {
+        const data = await getCurrentConditions(location);
+        return clearWeatherConditions.includes(data.weather);
+    } catch (err) {
+        context.log(`E3_GetIsClear encountered an error: ${err}`);
+        throw new Error(err);
+    }
+}
 
 function getCurrentConditions(location) {
-    return new Promise(function (resolve, reject) {
-        const options = {
-            url: `https://api.wunderground.com/api/${process.env["WeatherUndergroundApiKey"]}/conditions/q/${location.state}/${location.city}.json`,
-            method: 'GET',
-            json: true
-        };
-        request(options, function (err, res, body) {
-            if (err) {
-                reject(err);
-            }
+    const options = {
+        url: `https://api.wunderground.com/api/${process.env["WeatherUndergroundApiKey"]}/conditions/q/${location.state}/${location.city}.json`,
+        method: 'GET',
+        json: true
+    };
+
+    return request(options)
+        .then((body) => {
             if (body.error) {
                 reject(body.error);
-            }
-            if (body.response.error) {
+            } else if (body.response && body.response.error) {
                 reject(body.response.error);
+            } else {
+                resolve(body.current_observation);
             }
-            resolve(body.current_observation);
```

Comment:
You could make this async too, I generally think it's a bit cleaner!

Something like (although would recommend testing):
```javascript
let body = await request(options);
if (body.error) {
    throw body.error;
} else if (body.response && body.response.error) {
    throw body.response.error;
} else {
    return body.current_observation;
}
```

Reply:
Tested, and it works.